source/common/quic/client_connection_factory_impl.cc

```diff
@@ -19,19 +19,34 @@ getContext(Network::TransportSocketFactory& transport_socket_factory) {
   auto* quic_socket_factory =
       dynamic_cast<QuicClientTransportSocketFactory*>(&transport_socket_factory);
   ASSERT(quic_socket_factory != nullptr);
-  ASSERT(quic_socket_factory->sslCtx() != nullptr);
   return quic_socket_factory->sslCtx();
 }
 
+std::shared_ptr<quic::QuicCryptoClientConfig> PersistentQuicInfoImpl::cryptoConfig() {
```

Comment:
This is somewhat laid-back updating. If a client connection is created before the SSL context update, will its crypto config not be updated till another createQuicNetworkConnection() is called?

Reply:
I don't think once a quic client is created we support reloading, so AFIK it'd only be applied when we created a new client anyway.

Reply:
ACK. 
Another question is if we need to copy over resumption tickets stored in EnvoyQuicSessionCache before replacing crypto config?

Reply:
I would think if we changed client certs we'd not want to keep resumption info?  cc @RyanTheOptimist

Reply:
Yup, that sounds right to me.